src/main/java/laba/pages/iosPages/ProductsListPageIOS.java

```diff
@@ -0,0 +1,164 @@
+import com.zebrunner.carina.webdriver.locator.ExtendedFindBy;
+
+import laba.components.ios.IOSFooterComponent;
+import laba.components.ios.IOSHeaderMenuComponent;
+import laba.components.ios.IOSProductComponent;
+import laba.components.ios.IOSSideMenuComponent;
+import laba.constants.MenuButtons;
+import laba.model.Product;
+import laba.pages.basePages.DrawingPageBase;
+import laba.pages.basePages.LoginPageBase;
+import laba.pages.basePages.ProductDetailsPageBase;
+import laba.pages.basePages.ProductsListPageBase;
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = ProductsListPageBase.class)
+public class ProductsListPageIOS extends ProductsListPageBase {
+
+    @FindBy(xpath = "//XCUIElementTypeOther[@name='test-Menu']")
+    private IOSHeaderMenuComponent headerMenu;
+
+    @FindBy(xpath = "//XCUIElementTypeStaticText[@name='PRODUCTS']")
+    private ExtendedWebElement title;
+
+    @FindBy(xpath = "//XCUIElementTypeOther[@name='test-Item']")
+    private List<IOSProductComponent> productList;
+
+    @FindBy(xpath = "//XCUIElementTypeStaticText[contains(@name, 'Sauce Labs. All Rights Reserved')]")
+    private IOSFooterComponent footer;
+
+    @FindBy(xpath = "//XCUIElementTypeOther[@name='test-Close']")
+    private IOSSideMenuComponent sideMenu;
+
+    @ExtendedFindBy(accessibilityId = "test-Modal Selector Button")
+    private ExtendedWebElement sortDropdown;
+
+    @FindBy(xpath = "//XCUIElementTypeOther[@name='%s']")
+    private ExtendedWebElement sortOption;
+
+    public ProductsListPageIOS(WebDriver driver) {
+        super(driver);
+        setUiLoadedMarker(title);
+    }
+
+    @Override
+    public ProductDetailsPageBase openProductByName(String productName) {
+        for (IOSProductComponent product : productList) {
+            if (product.getProductName().equalsIgnoreCase(productName)) {
+                product.clickOnProductName();
+                return initPage(getDriver(), ProductDetailsPageBase.class);
+            }
+        }
+        throw new NoSuchElementException("Product not found: " + productName);
+    }
+
+    @Override
+    public List<String> getAllProductNames() {
+        List<String> names = new ArrayList<>();
+        for (IOSProductComponent item : productList) {
+            names.add(item.getProductName());
+        }
+        return names;
+    }
+
+    @Override
+    public List<Double> getAllProductPrices() {
+        List<Double> prices = new ArrayList<>();
+        for (IOSProductComponent item : productList) {
+            prices.add(Double.parseDouble(item.getProductPrice().replaceAll("[^\\d.]", "")));
+        }
+        return prices;
+    }
+
+    @Override
+    public IOSFooterComponent getFooter() {
+        return footer;
+    }
+
+    @Override
+    public IOSSideMenuComponent getSideMenu() {
+        return sideMenu;
+    }
+
+    @Override
+    public IOSHeaderMenuComponent getHeaderMenu() {
+        return headerMenu;
+    }
```

Comment:
To avoid duplicate code across multiple page classes, you can extract the shared components (footer, sideMenu, headerMenu)